wolfssl/wolfcrypt/types.h

```diff
@@ -495,11 +495,12 @@ decouple library dependencies with standard string, memory and so on.
 
         #define XSTRLEN(s1)       strlen((s1))
         #define XSTRNCPY(s1,s2,n) strncpy((s1),(s2),(n))
-        /* strstr, strncmp, and strncat only used by wolfSSL proper,
+        /* strstr, strncmp, strcmp, and strncat only used by wolfSSL proper,
          * not required for wolfCrypt only */
         #define XSTRSTR(s1,s2)    strstr((s1),(s2))
         #define XSTRNSTR(s1,s2,n) mystrnstr((s1),(s2),(n))
         #define XSTRNCMP(s1,s2,n) strncmp((s1),(s2),(n))
+        #define XSTRCMP(s1,s2)    strcmp((s1),(s2))
```

Comment:
This is used in the `WOLFSSL_CONF` part of the code where we are comparing two C strings. It would be the same as calling `XSTRLEN` twice, checking length, and comparing with `XSTRNCMP`. Using `XSTRCMP` makes the code cleaner and has the same result. I can refactor to avoid `XSTRCMP`.